hazelcast-jet-core/src/main/java/com/hazelcast/jet/core/processor/Processors.java

```diff
@@ -972,6 +976,22 @@ private Processors() {
                 (singletonTraverser, service, item) -> flatMapFn.apply(service, item));
     }
 
+    /**
+     * Returns a supplier of processors for a vertex that sorts its input using
+     * a {@link PriorityQueue} and emits it in the {@code complete} phase.
+     * <p>
+     * The output edge of this vertex should be {@link Edge#distributed
+     * distributed} {@link Edge#monotonicOrder monotonicOrder} {@link
+     * Edge#allToOne allToOne} so it preserves the ordering when merging
+     * the data from all upstream processors.
+     *
+     * @since 4.3
```

Comment:
- Can you explain which kind of input data this processor can effectively sort (It seems to me that it cannot sort the streaming data completely, but I'm not sure.)?
- Is this processor's only use case to sort the data completely or can it be used for other purposes such as partial sorting? 
- Also, could you document what the comparator parameter does and what the type parameter is?

Reply:
Also, we should change `V` to `T` for consistency.

Reply:
1- It's only meant for batch use cases, it's not meant to be available in `StreamingStage` interface.
2- `SortP` can only do partial sorting, global sorting is done by combining partially sorted data.

Reply:
I think we should add this information to this processor's javadoc.

Reply:
Actually, this inspires adding a new Edge property that's a limit on the number of items allowed to travel over it.
This simple change introduces partial sorting to Jet!
usage: 
`items.partialSort(10)`, 
`items.partialSort(ComparatorEx.reverseOrder(), 10)`

Reply:
I guess partial sorting could still be done using `stage.sort().aggregate(topN())` or `stage.sort().aggregate(bottomN())` but that means sending all data first to the cluster member that does global sorting to complete the sorting stage then choosing top or bottom N items from them.
I assume having an edge that has a limit on the number of items that can be sent over it would be a better solution, that's why I suggested using the new `stage.partialSort()`